model/host/host.go

```diff
@@ -1471,21 +1471,17 @@ func (h *Host) clearRunningTaskWithFunc(doUpdate func(update bson.M) error) erro
 // UpdateRunningTaskWithContext updates the running task for the host. It does
 // not log an event for task assignment.
 func (h *Host) UpdateRunningTaskWithContext(ctx context.Context, env evergreen.Environment, t *task.Task) error {
-	doUpdate := func(query, update bson.M) error {
-		_, err := env.DB().Collection(Collection).UpdateOne(ctx, query, update)
-		return err
-	}
-	return h.updateRunningTaskWithFunc(doUpdate, t)
-}
-
-func (h *Host) updateRunningTaskWithFunc(doUpdate func(query, update bson.M) error, t *task.Task) error {
 	if t == nil {
 		return errors.New("received nil task, cannot update")
 	}
 	if t.Id == "" {
 		return errors.New("task has empty task ID, cannot update")
 	}
 
+	doUpdate := func(query, update bson.M) error {
```

Comment:
We don't need this update functor anymore since we can just call `env.DB()...UpdateOne(...)` directly on L1509.